tests/ord-service/tests/subscription_flow_test.go

```diff
@@ -0,0 +1,184 @@
+	CustomerID             string
+	Subdomain              string
+	SubscriptionProviderID string
+}
+
+func TestSubscriptionFlow(t *testing.T) {
+	ctx := context.Background()
+	defaultTenantId := tenant.TestTenants.GetDefaultTenantID()
+	secondaryTenant := tenant.TestTenants.GetIDByName(t, tenant.ApplicationsForRuntimeTenantName)
+	subscriptionProviderID := "xs-app-name"
+	subscriptionConsumerID := "1f538f34-30bf-4d3d-aeaa-02e69eef84ae"
+	accountID := uuid.New().String()
+
+	runtimeInput := graphql.RuntimeInput{
+		Name:        "testingRuntime",
+		Description: ptr.String("testingRuntime-description"),
+		Labels:      graphql.Labels{testConfig.SubscriptionProviderLabelKey: subscriptionProviderID, "region": regionPathParamValue},
+	}
+
+	runtime, err := fixtures.RegisterRuntimeFromInputWithinTenant(t, ctx, dexGraphQLClient, defaultTenantId, &runtimeInput)
+	defer fixtures.CleanupRuntime(t, ctx, dexGraphQLClient, defaultTenantId, &runtime)
+	require.NoError(t, err)
+	require.NotEmpty(t, runtime.ID)
+
+	app, err := fixtures.RegisterApplication(t, ctx, dexGraphQLClient, "testingApp", secondaryTenant)
+	defer fixtures.CleanupApplication(t, ctx, dexGraphQLClient, secondaryTenant, &app)
+	require.NoError(t, err)
+	require.NotEmpty(t, app.ID)
+
+	consumerApp, err := fixtures.RegisterApplication(t, ctx, dexGraphQLClient, "consumerApp", secondaryTenant)
+	defer fixtures.CleanupApplication(t, ctx, dexGraphQLClient, secondaryTenant, &consumerApp)
+	require.NoError(t, err)
+	require.NotEmpty(t, consumerApp.ID)
+
+	// Create label definition
+	scenarios := []string{"DEFAULT", "consumer-test-scenario"}
+	fixtures.UpdateScenariosLabelDefinitionWithinTenant(t, ctx, dexGraphQLClient, secondaryTenant, scenarios)
+	defer fixtures.UpdateScenariosLabelDefinitionWithinTenant(t, ctx, dexGraphQLClient, secondaryTenant, scenarios[:1])
+
+	// Create automatic scenario assigment for consumer subaccount
+	asaInput := fixtures.FixAutomaticScenarioAssigmentInput(scenariosLabel, selectorKey, subscriptionConsumerID)
+	fixtures.CreateAutomaticScenarioAssignmentInTenant(t, ctx, dexGraphQLClient, asaInput, secondaryTenant)
+	defer fixtures.DeleteAutomaticScenarioAssignmentForScenarioWithinTenant(t, ctx, dexGraphQLClient, secondaryTenant, scenarioName)
+
+	// Set application scenarios label
+	fixtures.SetApplicationLabel(t, ctx, dexGraphQLClient, consumerApp.ID, scenariosLabel, scenarios[1:])
+	defer fixtures.SetApplicationLabel(t, ctx, dexGraphQLClient, consumerApp.ID, scenariosLabel, scenarios[:1])
+
+	providedTenant := Tenant{
+		TenantID:               accountID,
+		SubaccountID:           subscriptionConsumerID,
+		Subdomain:              defaultSubdomain,
+		SubscriptionProviderID: subscriptionProviderID,
+	}
+
+	// Build a request for consumer subscription
+	request := createTenantRequest(t, providedTenant, http.MethodPut, testConfig.TenantFetcherFullRegionalURL)
+
+	t.Log(fmt.Sprintf("Provisioning tenant with ID %s", actualTenantID(providedTenant)))
```

Comment:
A little misleading log in the context of this test. I think something along the lines of:
`Creating a subscription between consumer with id %s and provider with name %s`
will be better.